src/core/vector/qgsvectorlayertemporalproperties.cpp

```diff
@@ -554,23 +546,19 @@ QString QgsVectorLayerTemporalProperties::createFilterString( const QgsVectorLay
     {
       if ( !mStartExpression.isEmpty() && !mEndExpression.isEmpty() )
       {
-        return QStringLiteral( "((%1) %2 %3) AND ((%4) %5 %6)" ).arg( mStartExpression,
-               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
+        return QStringLiteral( "((%1) < %2) AND ((%3) > %4)" ).arg( mStartExpression,
                dateTimeExpressionLiteral( range.end() ),
                mEndExpression,
-               range.includeBeginning() ? QStringLiteral( ">=" ) : QStringLiteral( ">" ),
                dateTimeExpressionLiteral( range.begin() ) );
       }
       else if ( !mStartExpression.isEmpty() )
       {
-        return QStringLiteral( "(%1) %2 %3" ).arg( mStartExpression,
```

Comment:
I think the logic here is broken (completely in the current code!)

I believe the correct one is:

```
return QStringLiteral( "(%1) %2 %3" ).arg( mStartExpression,
    range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
    dateTimeExpressionLiteral( range.end() ) );
```